projectq/backends/_awsbraket/_awsbraket_boto3_client.py

```diff
@@ -0,0 +1,502 @@
+#   Copyright 2021 ProjectQ-Framework (www.projectq.ch)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Back-end to run quantum program on AWS Braket supported devices.
+
+This backend requires the official AWS SDK for Python, Boto3.
+The installation is very simple
+> pip install boto3
+"""
+
+import getpass
+import signal
+import re
+import time
+import boto3
+import botocore
+
+import json
+
+
+class AWSBraket():
+    """
+    Manage a session between ProjectQ and AWS Braket service.
+    """
+    def __init__(self):
+        self.backends = dict()
+        self.timeout = 5.0
+        self._credentials = []
+        self._s3_folder = []
+
+    def _authenticate(self, credentials=None):
+        """
+        Args:
+            credentials (list): contains the AWS_ACCESS_KEY and AWS_SECRET_KEY.
+                    and aws_secret_access_key
+        """
+        if credentials is None:
+            aws_acess_key = getpass.getpass(prompt="Enter AWS_ACCESS_KEY_ID: ")
+            aws_secret_key = getpass.getpass(prompt="Enter AWS_SECRET_KEY: ")
+            credentials = [aws_acess_key, aws_secret_key]
+
+        self._credentials = credentials
+
+    def _get_s3_folder(self, s3_folder=None):
+        """
+        Args:
+           s3_folder (list): contains the S3 bucket and directory
+           to store the results.
+        """
+        if s3_folder is None:
+            S3Bucket = input("Enter the S3 Bucket configured in Braket: ")
+            S3Directory = input(
+                "Enter the Directory created in the S3 Bucket: ")
+            s3_folder = [S3Bucket, S3Directory]
+
+        self._s3_folder = s3_folder
+
+    def get_list_devices(self, verbose=False):
+        """
+        Get the list of available devices with their basic properties
+
+        Args:
+            verbose (bool): print the returned dictionnary if True
+
+        Returns:
+            (dict) backends dictionary by deviceName, containing the qubit
+                   size 'nq', the coupling map 'coupling_map' if applicable
+                   (IonQ device as an ion device is having full connectivity)
+                   and the Schema Header version 'version', because it seems
+                   that no device version is available by now
+        """
+        # TODO: refresh region_names if more regions get devices available
```

Comment:
Unfortunatelly it seems that it is not implemented yet. There is a way to obtain the available regions for an AWS service in general, but it is not implemented in braket.

> import boto3
> s = boto3.Session()
> d_reg = s.get_available_regions('dynamodb')
> print(d_reg)
['af-south-1', 'ap-east-1', 'ap-northeast-1', 'ap-northeast-2', 'ap-south-1', 'ap-southeast-1', 'ap-southeast-2', 'ca-central-1', 'eu-central-1', 'eu-north-1', 'eu-south-1', 'eu-west-1', 'eu-west-2', 'eu-west-3', 'me-south-1', 'sa-east-1', 'us-east-1', 'us-east-2', 'us-west-1', 'us-west-2']

> b_reg = s.get_available_regions('braket')
> print(c_reg)
[]

I sugest to wait until this function is implemented in boto3, because most probably there will be not much change in the available regions for braket.

Reply:
Ok, then leave your comment as-is and we can always return to it in the future once the functionality has been implemented on their side.